analysis/preprocess/preprocess_data.R

```diff
@@ -22,19 +22,29 @@ fs::dir_create(here::here("output", "review"))
 
 input_path<-paste0("output/input_",cohort_name,".csv.gz")
 
+
+
 # # Get colnames 
 col_names <- fread(input_path, header = TRUE, sep = ",", nrows = 1, stringsAsFactors = FALSE)
 
-#Get column with "_date"
+#Get columns types based on their names. 
```

Comment:
Hi Marwa - this is great! I wonder if we should assign all columns as there are only 9 left after you have done this. Potential code below:

```
# Get colnames 

cols <- colnames(fread(input_path, header = TRUE, sep = ",", nrows = 0, stringsAsFactors = FALSE))

# Get columns types based on their names

date_cols <- grep("_date", cols, value = TRUE)

cat_cols <- c("patient_id",
              grep("_cat", cols, value=TRUE))

bin_cols <- c(grep("_bin", cols, value=TRUE), 
                   grep("prostate_cancer_", cols, value=TRUE),
                   "has_follow_up_previous_6months","has_died","registered_at_start")
              
num_cols <- c(grep("_num", cols, value=TRUE),
              grep("vax_jcvi_age_", cols, value=TRUE))

# Set the class of the columns 

col_classes <- setNames(rep("Date", length(out_date_cols)), date_cols)
col_classes <- setNames(rep("character", length(cat_cols)), cat_cols)
col_classes <- setNames(rep("logical", length(bin_cols)), bin_cols)
col_classes <- setNames(rep("d", length(num_cols)), num_cols)
```

Reply:
Agree, I also removed cov_num_systolic_bp_date_measured because it was causing parsing issues as it has both _date and _num